packages/editor/src/lib/hooks/useScreenBounds.ts

```diff
@@ -1,22 +1,35 @@
 import throttle from 'lodash.throttle'
 import { useLayoutEffect } from 'react'
+import { Box } from '../primitives/Box'
 import { useEditor } from './useEditor'
 
-export function useScreenBounds() {
+export function useScreenBounds(ref: React.RefObject<HTMLElement>) {
 	const editor = useEditor()
 
 	useLayoutEffect(() => {
-		const updateBounds = throttle(
-			() => {
-				editor.updateViewportScreenBounds()
-			},
-			200,
-			{
-				trailing: true,
+		const updateScreenBounds = () => {
+			const container = ref.current
+			if (!container) {
+				return null
 			}
-		)
 
-		editor.updateViewportScreenBounds()
+			const rect = container.getBoundingClientRect()
+
+			editor.updateViewportScreenBounds(
+				new Box(
+					rect.left || rect.x,
+					rect.top || rect.y,
+					Math.max(rect.width, 1),
+					Math.max(rect.height, 1)
+				)
+			)
+		}
+
+		const updateBounds = throttle(updateScreenBounds, 200, {
+			trailing: true,
+		})
+
+		updateScreenBounds()
```

Comment:
maybe out of scope of this change, but as we're doing more work on allowing the tldraw component to be used outside of full-screen contexts, we could think about using a `ResizeObserver` here too

Reply:
That's a good shout! I thought we did have one here. I'll add.